lib/internal/bootstrap_node.js

```diff
@@ -127,18 +127,11 @@
 
         // check if user passed `-c` or `--check` arguments to Node.
         if (process._syntax_check_only != null) {
-          const vm = NativeModule.require('vm');
           const fs = NativeModule.require('fs');
-          const internalModule = NativeModule.require('internal/module');
           // read the source
           const filename = Module._resolveFilename(process.argv[1]);
           var source = fs.readFileSync(filename, 'utf-8');
-          // remove shebang and BOM
-          source = internalModule.stripBOM(source.replace(/^#!.*/, ''));
-          // wrap it
-          source = Module.wrap(source);
-          // compile the script, this will throw if it fails
-          new vm.Script(source, {filename: filename, displayErrors: true});
```

Comment:
If I'm not mistaken, the filename doesn't affect whether a script is syntactically valid, and piped-in code has no filename.